### PR TITLE
feat(logger): add removeKeys functionality

### PR DESCRIPTION
**Pull request by saragerion**
## Description of your changes

This PR adds the option of removing persistent keys from a logger instance. 
Developers are now able to do it using the method `removeKeys`. The method accepts as parameter an array of keys to be removed.

carbon (2)

### How to verify this change

Unit tests & E2E tests added and passing locally.

### Related issues, RFCs

#482 

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "test": "npm run lerna-test",
     "commit": "commit",
     "package": "npm run package",
+    "setup-local": "npm ci --foreground-scripts && cd examples/cdk && npm ci && cd ../.. && cd examples/sam && npm ci && cd ../..",
```
Comment (saragerion) [inline package.json:18]: Sorry, I made this out of laziness :-)

Comment (dreamorosi) [reply package.json:18]: Maybe we could mention this in this section of the docs? https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/CONTRIBUTING.md?plain=1#L81

Happy to do it in a separate PR if you want, but I think it's great that we have this command. Someone mentioned this in an issue also in the past.

File: packages/logger/tests/unit/Logger.test.ts [hunk 2]
```diff
@@ -638,7 +638,7 @@ describe('Class: Logger', () => {
 
   describe('Method: appendKeys', () => {
 
-    test('when called, populates the logger\'s propriety persistentLogAttributes ', () => {
+    test('when called, it populates the logger\'s propriety persistentLogAttributes ', () => {
```
Comment (dreamorosi) [inline packages/logger/tests/unit/Logger.test.ts:641]:
```suggestion
    test('when called, it populates the logger\'s persistentLogAttributes property', () => {
```

File: packages/logger/tests/unit/Logger.test.ts [hunk 3]
```diff
@@ -704,6 +704,97 @@ describe('Class: Logger', () => {
     });
   });
 
+  describe('Method: removeKeys', () => {
+
+    test('when called, it removes keys from the logger\'s propriety persistentLogAttributes', () => {
```
Comment (dreamorosi) [inline packages/logger/tests/unit/Logger.test.ts:709]:
```suggestion
    test('when called, it removes keys from the logger\'s persistentLogAttributes property', () => {
```

**Review by dreamorosi — APPROVED** [review]
Thanks for the PR, looking great - left only a couple of minor comments about typos.

PS: very nice to review a PR from you again 😄

**Review by dreamorosi — APPROVED** [review]

**Review by flochaz — APPROVED** [review]

**Pull request merged**